tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetRepository.scala

```diff
@@ -0,0 +1,58 @@
+package com.linagora.tmail.james.jmap.publicAsset
+
+import com.google.common.collect.{HashBasedTable, Table, Tables}
+import org.apache.james.core.Username
+import org.apache.james.jmap.api.model.IdentityId
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import scala.jdk.CollectionConverters._
+
+trait PublicAssetRepository {
+  def create(username: Username, creationRequest: PublicAsset): Publisher[PublicAsset]
+
+  def update(username: Username, id: PublicAssetId, identityIds: Set[IdentityId]): Publisher[Void]
+
+  def remove(username: Username, id: PublicAssetId): Publisher[Void]
+
+  def revoke(username: Username): Publisher[Void]
+
+  def get(username: Username, ids: Set[PublicAssetId]): Publisher[PublicAsset]
+
+  def get(username: Username, id: PublicAssetId): Publisher[PublicAsset] = get(username, Set(id))
+
+  def list(username: Username): Publisher[PublicAsset]
+
+}
+
+class MemoryPublicAssetRepository extends PublicAssetRepository {
```

Comment:
A good way to stucture well the repsitory is to force ourselves to use the `BlobStore` to store the content.